tests/Compat/CompatFormTest.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace ipl\Tests\Web\Compat;
+
+use ipl\Html\FormElement\SubmitElement;
+use ipl\Tests\Html\TestCase;
+use ipl\Web\Compat\CompatForm;
+
+class CompatFormTest extends TestCase
+{
+    /** @var CompatForm */
+    private $form;
+
+    protected function setUp(): void
+    {
+        $this->form = new CompatForm();
+    }
+
+    public function testDuplicateSubmitButtonApplied(): void
+    {
+        $this->form->addElement('submit', 'submitCreate');
+        $this->form->addElement('submit', 'submitDelete');
+
+        $expected = <<<'HTML'
+    <form class="icinga-form icinga-controls" method="POST">
+      <input class="primary-submit-btn-duplicate" name="submitCreate" type="submit" value="submitCreate"/>
+      <div class="control-group form-controls">
+        <input class="btn-primary" name="submitCreate" type="submit" value="submitCreate"/>
+      </div>
+      <div class="control-group form-controls">
+        <input class="btn-primary" name="submitDelete" type="submit" value="submitDelete"/>
+      </div>
+    </form>
+HTML;
+
+        $this->assertHtml($expected, $this->form);
+    }
+
+    public function testDuplicateSubmitButtonOmitted(): void
+    {
+        $this->form->addElement('submit', 'submitCreate');
+
+        $expected = <<<'HTML'
+    <form class="icinga-form icinga-controls" method="POST">
+      <div class="control-group form-controls">
+        <input class="btn-primary" name="submitCreate" type="submit" value="submitCreate"/>
+      </div>
+    </form>
+HTML;
+
+        $this->assertHtml($expected, $this->form);
+    }
+
+    public function testDuplicateSubmitButtonAddedOnlyOnce(): void
+    {
+        $this->form->addElement('submit', 'submitCreate', ['id' => 'submit_id']);
+        $this->form->addElement('submit', 'submitDelete');
+
+        $expected = <<<'HTML'
+    <form class="icinga-form icinga-controls" method="POST">
+      <input class="primary-submit-btn-duplicate" name="submitCreate" type="submit" value="submitCreate"/>
+      <div class="control-group form-controls">
+        <input id="submit_id" class="btn-primary" name="submitCreate" type="submit" value="submitCreate"/>
+      </div>
+      <div class="control-group form-controls">
+        <input class="btn-primary" name="submitDelete" type="submit" value="submitDelete"/>
+      </div>
+    </form>
+HTML;
+
+        $this->form->render();
```

Comment:
Why the render call?

Reply:
To render it once, and again in the `assertHtml()`. To make sure it only prepends one submit button.

Reply:
I would add  something like this as comment before the render call and remove the newline before `asserHtml()` as this block is the test case, i.e.

```php
// Submit button is duplicated in render(), so we have to call it twice for the test.
render()
assert()